### ISSUE TITLE
feat: LLM inference libraries support plan

### ISSUE THREAD

**Issue by hydai** [WasmEdge/WasmEdge#3124]
### Summary

There are various LLM inference libraries. WasmEdge already integrated llama.cpp, but we want to bring more to the community.

### Details

Already supported:

1. PyTorch
2. TFLite
3. OpenVINO

The support priority list:
- Tier 1:
  - burn-rs
- Tier 2:
  - Intel extension for transformers
  - mlx
  - whisper.cpp
- Tier 3:
  - vllm
  - CTranslate2
  - candle

Please feel free to add any comments and suggestions. We would like to hear the voice of the community.
Also, if you are interested in contributing the new LLM inference libraries, please tell and show to us :-)

Happy new year!

**Comment by katopz** [comment]
Cool, i'm now playing with
- For RAG: [llm-chain](https://github.com/sobelio/llm-chain) // just support Qdrant
- For mac: mlx // still in heavy development
- For fun: candle // infer tps seem to be a lot slower than `llama.cpp` atm but [it will keep up somehow i hope](https://twitter.com/reach_vb/status/1740804591095283899).
- For co-pilot: [Tabby](https://github.com/TabbyML/tabby) // Support repos lookup but the output not satisfy me yet

**Comment by akashicMarga** [comment]
Any updates/discussions on other backends like burn and mlx? i see burn commit merged. @katopz

**Comment by hydai** [comment]
Initial support of burn is merged.
MLX is ongoing, there is an existing PR on it.

**Comment by ShigrafS** [comment]
Hi @hydai , just wanted to confirm if this is correct: burn is merged, MLX is in progress, and RWKV, vLLM, CTranslate2, and candle are still pending? 
Please let me know if anything's changed.
Thanks!

**Comment by hydai** [comment]
> Hi [@hydai](https://github.com/hydai) , just wanted to confirm if this is correct: burn is merged, MLX is in progress, and RWKV, vLLM, CTranslate2, and candle are still pending?

Yup, that's correct. Any contributions to these new backends are welcome.

**Comment by ShigrafS** [comment]
@hydai 
I would like to work on `RWKV` integration.

**Comment by hydai** [comment]
Sure, please go ahead. Thank you.

**Comment by PhantomInTheWire** [comment]
@hydai i think i would like to attempt RWKV too, anything important i should keep in mind for this?

**Comment by hydai** [comment]
Hi @PhantomInTheWire,  
Please ensure you are using the latest RWKV. It would also be better if you could verify the behavior before submitting a PR, as waiting for our approval to run these LLM inferences is time-consuming.

**Comment by PhantomInTheWire** [comment]
hey @hydai i agree i'll test everything e2e and make sure it works before submitting a pr

**Comment by hydai** [comment]
Closing this because, due to the capacity of the maintainers, we no longer want to add new backend support for WASI-NN.
